src/dlgTriggerEditor.cpp

```diff
@@ -125,68 +119,56 @@ dlgTriggerEditor::dlgTriggerEditor( Host * pH )
     setStatusBar( statusBar );
     statusBar->show();
     mIsGrabKey = false;
-    QVBoxLayout * pVB1 = new QVBoxLayout(mainArea);
+
+    // Replacement code to use single splitter for all right widgets:
+    mpSystemMessageArea = new dlgSystemMessageArea(splitter_rightTopBottom);
+    mpTriggersMainArea = new dlgTriggersMainArea(splitter_rightTopBottom);
+    mpTimersMainArea = new dlgTimersMainArea(splitter_rightTopBottom);
+    mpAliasMainArea = new dlgAliasMainArea(splitter_rightTopBottom);
+    mpActionsMainArea = new dlgActionMainArea(splitter_rightTopBottom);
+    mpKeysMainArea = new dlgKeysMainArea(splitter_rightTopBottom);
+    mpVarsMainArea = new dlgVarsMainArea(splitter_rightTopBottom);
+    mpScriptsMainArea = new dlgScriptsMainArea(splitter_rightTopBottom);
+    mpSourceEditorArea = new dlgSourceEditorArea(splitter_rightTopBottom);
+    mpErrorConsole = new TConsole(mpHost,false,splitter_rightTopBottom);
+    mpErrorConsole->setMinimumHeight( 200 );
+
+    //Prevent any region being "squashed down" i.e. taken below its minimum size
+    //so that it "disappears"
+    splitter_rightTopBottom->setChildrenCollapsible(false);
+    splitter_main->setChildrenCollapsible(false);
```

Comment:
I agree with you, we need some mechanism other than dragging the splitter back, because that's not always an intuitive way out.

Let's think this through. I think the best solution would be one that's intuitive (duh) and in a way re-uses the same mechanics of people already know.

Errors view: should be able to restore a splitter that's set to minimum by pressing the errors view button again. This'll work because the button is an on/off toggle for errors already, so people see that the errors is missing (even though it's there), press it and it shows.

Item hierarchy vertical splitter: I'm not sure about changing item type, because it would be nice to have the splitter go back to normal once you switch to your original view - let's say the use case is that I navigate away from editing my trigger "fullscreen" to see some aliases and want to come back to my trigger to continuing editing it. Need to think, have to attend a meeting.

Item content horizontal splitter: need to think.